retrofit/src/main/java/retrofit/MethodHandler.java

```diff
@@ -28,7 +29,9 @@
     Converter<ResponseBody, Object> responseConverter =
         (Converter<ResponseBody, Object>) createResponseConverter(method, retrofit, responseType);
     RequestFactory requestFactory = RequestFactoryParser.parse(method, responseType, retrofit);
-    return new MethodHandler<>(retrofit, requestFactory, callAdapter, responseConverter);
+    String infoForException = collectInfoForException(method);
```

Comment:
If it's too difficult, I at least think this belongs inside `RequestFactoryParser` and becomes `.describe()` or `.toString()` on the `RequestFactory` object.